frameworks/cassandra/src/main/dist/svc.yml

```diff
@@ -146,19 +298,51 @@ pods:
         {{/TASKCFG_ALL_SECURITY_TRANSPORT_ENCRYPTION_ENABLED}}
       snapshot:
         goal: ONCE
-        cmd: >
-                export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jdk*/) ;
-                ./apache-cassandra-{{CASSANDRA_VERSION}}/bin/nodetool snapshot -p ${JMX_PORT} -t "$SNAPSHOT_NAME" -- $(eval "echo $CASSANDRA_KEYSPACES")
+        cmd:  |
+                export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jdk*/) 
+                {{#TASKCFG_ALL_SECURE_JMX_ENABLED}}
+                ./bootstrap --resolve=false
+                cd $MESOS_SANDBOX
+                export HOME=$MESOS_SANDBOX
+                export SECURE_JMX_KEY_STORE_KEY=`cat jmx/key_store_pass`
+                export KEY_STORE_PATH=jmx/key_store
+                if [ -f "jmx/trust_store" ]; then
```

Comment:
we are using a lot of repeated code in commands, for example, this piece of code block 
```
export JAVA_HOME=$(ls -d $MESOS_SANDBOX/jdk*/)
{{#TASKCFG_ALL_SECURE_JMX_ENABLED}}
./bootstrap --resolve=false
cd $MESOS_SANDBOX
export HOME=$MESOS_SANDBOX
export SECURE_JMX_KEY_STORE_KEY=`cat jmx/key_store_pass`
export KEY_STORE_PATH=jmx/key_store
if [ -f "jmx/trust_store" ]; then
    # load user provided trust store
    export TRUST_STORE_PATH=jmx/trust_store
    export SECURE_JMX_TRUST_STORE_KEY=`cat jmx/trust_store_pass`
else
    # load user provided keystore certs to default truststore
    export TRUST_STORE_PATH=$JAVA_HOME/lib/security/cacerts
    export SECURE_JMX_TRUST_STORE_KEY="changeit"
    ${JAVA_HOME}/bin/keytool -noprompt -importkeystore -srckeystore $KEY_STORE_PATH -srcstorepass $SECURE_JMX_KEY_STORE_KEY -destkeystore $TRUST_STORE_PATH -deststorepass $SECURE_JMX_TRUST_STORE_KEY
fi
mkdir jmx_properties
mkdir .cassandra
(umask u=rw,g=,o= && cp jmx/key_file jmx_properties/key_file)
chmod +x .ssl/jmx-ssl-properties.sh
.ssl/jmx-ssl-properties.sh
user=$(sed -nr '0,/readwrite/s/.*(^)(.+) readwrite.*/\2/p' $MESOS_SANDBOX/jmx/access_file)
```
is used 6 times, for the sake of maintainability I think we should move the repeated code to a script and call the script from everywhere its needed.

Reply:
+1

Reply:
added repeated code to jmx-ssl-setup.sh.